lib/coverband/utils/file_groups.rb

```diff
@@ -17,29 +18,36 @@ def grouped_results
 
       private
 
-      def filter_to_groups(files)
+      def filter_to_groups
         grouped_files = []
-        grouped_gems = {}
-        gem_lists = []
         Coverband.configuration.groups.each do |name, filter|
           if name == 'Gems'
-            grouped_gems = files.select { |source_file| source_file.filename =~ /#{filter}/ }.group_by(&:gem_name)
-            gem_lists = grouped_gems.values.map { |gem_files| Coverband::Utils::FileList.new(gem_files) }
-            grouped_files.concat(gem_lists.flatten)
-            @grouped[name] = Coverband::Utils::GemList.new(gem_lists) if gem_lists.flatten.any?
+            gem_lists = gem_files(name, filter)
+            grouped_files.concat(gem_lists.flatten) if gem_lists.flatten.any?
           else
-            @grouped[name] = Coverband::Utils::FileList.new(files.select do |source_file|
-              source_file.filename =~ /#{filter}/
-            end)
+            app_files(name, filter)
             grouped_files += @grouped[name]
           end
         end
-        if !Coverband.configuration.groups.empty? && !(other_files = files.reject do |source_file|
+        if !Coverband.configuration.groups.empty? && !(other_files = @files.reject do |source_file|
```

Comment:
This method is looking so much simpler! I still feel a little overwhelmed when looking at these last two lines. Perhaps there's a way we can refactor so it's more self documenting? Feel free to disregard as this is definitely ready to :ship.